### ISSUE TITLE
Metadata Panel doesn't close when settings panel clicked

### ISSUE THREAD

**Issue by rashidakanchwala** [kedro-org/kedro-viz#1261]
## Description

Metadata panel usually closes when you do any other action on kedro-viz. However it doesn't close when you try to change any of these (Hide label, Export Viz, or Hide Layers)

I am not sure if this is a bug or this is by design but I thought I would raise it anyway and it can be closed if it's the latter.

**Comment by tynandebold** [comment]
Good find. I don't think this is a bug. Any of the icons you click on the toolbar don't close the metadata panel. Maybe it's inconsistent though.

@comym @Mackay031 can you help us here please?

**Comment by comym** [comment]
Hi, when checking the Demo right now, I've noticed that the metadata side panel actually does close when interacting with the Filters.

Regardless of what it is now, I do think the side panel should only close:

- When you click close on the X icon on the panel
- When you click anywhere in the background of the flowchart
- When you click to change to a different registry pipeline by using the dropdown
- When switching to another section (Experiment Tracking)

I've marked it in red below.

When interacting with anything else like navigating the tree nav, filters, modals, minimap, label toggling, etc, the metadata side panel should be open.

![Screenshot 2023-02-17 at 13 37 22](https://user-images.githubusercontent.com/55446576/219667036-28cf88b1-0f4f-4a8c-9857-c7805797acb8.png)
